src/App.tsx

```diff
@@ -14,11 +15,115 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+
+type State = {
+  goodsList : string[],
+  visibility : boolean,
+  lengthLimit : number,
+};
+
+class App extends React.Component <Props, State> {
+  state = {
+    goodsList: [...goodsFromServer],
+    visibility: false,
+    lengthLimit: 1,
+  };
+
+  visibilitySwitch = () => {
+    this.setState((prevState) => ({
+      visibility: !prevState.visibility,
+    }));
+  };
+
+  buttonGenerator = (text:string, method:() => void) => (
+    <button
+      type="button"
+      className="button is-info is-light is-outlined is-rounded"
+      onClick={method}
+    >
+      {text}
+    </button>
+  );
+
+  reverseList = () => {
+    this.setState((prevState) => ({
+      goodsList: [...prevState.goodsList].reverse(),
+    }));
+  };
+
+  sortABC = () => {
+    this.setState((prevState) => ({
+      goodsList: [...prevState.goodsList].sort(
+        (good1, good2) => good1.localeCompare(good2),
+      ),
+    }));
+  };
+
+  initialOrder = () => {
+    this.setState(() => ({
+      goodsList: [...goodsFromServer],
+      lengthLimit: 1,
+    }));
+  };
+
+  byLength = () => {
+    this.setState((prevState) => ({
+      goodsList: [...prevState.goodsList].sort(
+        (good1, good2) => good1.length - good2.length,
+      ),
+    }));
+  };
+
+  changeLimit = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState(() => ({
+      lengthLimit: +event.target.value,
+      goodsList: [...goodsFromServer]
+        .filter(good => good.length >= +event.target.value),
+    }));
+  };
+
+  selectGenerator = (limit:number) => {
```

Comment:
I think it's better to remove this function and add this to the render method, cause it became hard to understand everything straight away